pageserver/src/tenant/secondary.rs

```diff
@@ -110,6 +110,27 @@ impl SecondaryTenant {
     fn get_tenant_shard_id(&self) -> &TenantShardId {
         &self.tenant_shard_id
     }
+
+    /// For API access: generate a LocationConfig equivalent to the one that would be used to
+    /// create a Tenant in the same state.  Do not use this in hot paths: it's for relatively
+    /// rare external API calls, like a reconciliation at startup.
+    pub(crate) fn get_location_conf(&self) -> models::LocationConfig {
+        let conf = self.detail.lock().unwrap().config.clone();
+
+        let conf = models::LocationConfigSecondary { warm: conf.warm };
+
+        models::LocationConfig {
+            mode: models::LocationConfigMode::Secondary,
+            generation: None,
+            secondary_conf: Some(conf),
+            shard_number: self.tenant_shard_id.shard_number.0,
+            shard_count: self.tenant_shard_id.shard_count.0,
+            // Shard stripe size is not set in secondary mode
+            shard_stripe_size: u32::default(),
+            // TenantConfig is for attached tenants, not set in secondary mode
+            tenant_conf: models::TenantConfig::default(),
```

Comment:
Maybe it should be an Option?

Reply:
I've gone back and forward on this... I'm also tempted to just make secondary locations store the tenantconfig.  They already persist it on disk, it just isn't carried in the SecondaryTenant.

Even if we make these fields Option<>, it feels... awkward.  I'm going to see how onerous it is to just carry the values around in SecondaryTenant.

Reply:
So yeah, it's way simpler to reason about if we just carry the values around so that we don't have to treat  them as optional.  I've done that in 1d2ed76539d5a39584f9e7ebdc16a8dd7d92fcca